Source/Extensions/Blazorise.DataGrid/DataGrid.razor.cs

```diff
@@ -2213,10 +2366,15 @@ public IEnumerable<GroupContext<TItem>> DisplayGroupedData
     [Parameter] public EventCallback<DataGridReadDataEventArgs<TItem>> ReadData { get; set; }
 
     /// <summary>
-    /// Occurs after the column sort direction has changed.
+    /// Occurs after the sort direction of a single column has changed.
     /// </summary>
     [Parameter] public EventCallback<DataGridSortChangedEventArgs> SortChanged { get; set; }
 
+    /// <summary>
+    /// Occurs when the sort order, sort direction, or number of columns to sort changes.
+    /// </summary>
+    [Parameter] public EventCallback<DataGridSortingChangedEventArgs> SortingChanged { get; set; }

```

Comment:
We have two similar named APIs. `SortChanged` and `SortingChanged`. What is the difference?

Reply:
I think I had this question myself before. Let me look it up, see if @danm-de has clarified it before.

Reply:
I couldn't find anything about this. But from what I can tell it's slightly different.
`SortChanged` is invoked on a single sorting.
`SortingChanged` is invoked when various sorting is applied.

![image](https://github.com/Megabit/Blazorise/assets/22283161/2534251f-fe96-47c1-848b-1a54b76ff393)

So on `SortOrder` you find specific single column attributes:
![image](https://github.com/Megabit/Blazorise/assets/22283161/413b2679-9062-472b-a900-fa2842b51822)
It still triggers the same as before, on `HandleSortColumn`.

On `SortingOrder` you find multiple column attributes:
![image](https://github.com/Megabit/Blazorise/assets/22283161/be96cf42-56e6-4d50-bb5c-fa1f6d3e73f2)
It triggers on:
- `HandleSortColumn`
- `ApplySorting`
- `RemoveColumn`

Altought I'm thinking, if we really need this event... 
- On `ApplySorting` seems redudant, since it's an api explicitly called. (Altought there can be a benefit when used in generic places.)
- On `RemoveColumn`, kinda makes sense, that if a sort column gets removed, a notification happens, but this seems like an off case and could be handled by the regular `SortChanged`
- `HandleSortColumn` -  Redudant...Can be benefical since it notifies of all the columns sort state...

What are your thoughts?

Reply:
> On ApplySorting seems redudant, since it's an api explicitly called. (Altought there can be a benefit when used in generic places.)

I think we can leave this. It is still good to have.

> On RemoveColumn, kinda makes sense, that if a sort column gets removed, a notification happens, but this seems like an off case and could be handled by the regular SortChanged

Makes sense to call `SortChanged` when column is removed since it can and will affect the query that the user build for their Rest APIs.

> HandleSortColumn - Redudant...Can be benefical since it notifies of all the columns sort state...

Not sure what you mean by this. Be already had it before.

Reply:
> Makes sense to call SortChanged when column is removed since it can and will affect the query that the user build for their Rest APIs.

User should get correct sort columns whenever `ReadData` is called in the `ReadData` arguments. They usually don't base themselves on this event, I assume...
Anyway I think it's fine. But again, the event called is not the existing `SortChanged`, but the new `SortingChanged` (Maybe it would make sense to call the existing one also actually)

What we are considering here, is whether `SortingChanged` is an unrelevant redudant event to be called. Can lead to user confusion, `SortChanged` ... `SortingChanged` I just got myself, I had to go check the code when you mentioned these.

Reply:
The users will definitely be confused. I mean, both of us are already confused. What would happen in half a year from now?

If possible there should only be one event.